GeoLib/IO/XmlIO/Qt/XmlGmlInterface.cpp

```diff
@@ -70,41 +71,66 @@ int XmlGmlInterface::readFile(const QString &fileName)
             {
                 ERR("XmlGmlInterface::readFile(): <name>-tag is empty.")
                 deleteGeometry(std::move(points), std::move(polylines),
-                               std::move(surfaces), pnt_names, ply_names,
-                               sfc_names);
+                               std::move(surfaces));
                 return 0;
             }
             else
                 gliName = type_node.toElement().text().toStdString();
         else if (nodeName.compare("points") == 0)
         {
-            readPoints(type_node, points.get(), pnt_names);
-            _geo_objs.addPointVec(std::move(points), gliName, pnt_names);
+            readPoints(type_node, points.get(), pnt_names.get());
+
+            // if names-map is empty, set it to NULL because it is not needed
+            if (pnt_names->empty())
+            {
+                pnt_names.reset(nullptr);
+            }
+            _geo_objs.addPointVec(std::move(points), gliName,
+                                  std::move(pnt_names));
         }
         else if (nodeName.compare("polylines") == 0)
+        {
             readPolylines(
                 type_node, polylines.get(), *_geo_objs.getPointVec(gliName),
-                _geo_objs.getPointVecObj(gliName)->getIDMap(), ply_names);
+                _geo_objs.getPointVecObj(gliName)->getIDMap(), ply_names.get());
+
+            // if names-map is empty, set it to NULL because it is not needed
+            if (ply_names->empty())
+            {
+                ply_names.reset(nullptr);
+            }
+        }
         else if (nodeName.compare("surfaces") == 0)
+        {
             readSurfaces(
                 type_node, surfaces.get(), *_geo_objs.getPointVec(gliName),
-                _geo_objs.getPointVecObj(gliName)->getIDMap(), sfc_names);
+                _geo_objs.getPointVecObj(gliName)->getIDMap(), sfc_names.get());
+
+            // if names-map is empty, set it to NULL because it is not needed
+            if (sfc_names->empty())
+            {
+                sfc_names.reset(nullptr);
+            }
+        }
     }
 
     if (polylines->empty())
-        deletePolylines(std::move(polylines), ply_names);
+        deletePolylines(std::move(polylines));
     else
-        _geo_objs.addPolylineVec(std::move(polylines), gliName, ply_names);
+        _geo_objs.addPolylineVec(std::move(polylines), gliName,
+                                 std::move(ply_names));
 
     if (surfaces->empty())
-        deleteSurfaces(std::move(surfaces), sfc_names);
+        deleteSurfaces(std::move(surfaces));
```

Comment:
There aren't any `Surface` objects in the vector `surfaces`. Is it necessary to call `deleteSurfaces`?